base_layer/wallet/src/lib.rs

```diff
@@ -41,9 +44,12 @@ use crate::{
     storage::sqlite_db::WalletSqliteDatabase,
     transaction_service::storage::sqlite_db::TransactionServiceSqliteDatabase,
 };
+
 pub type WalletSqlite = Wallet<
     WalletSqliteDatabase,
     TransactionServiceSqliteDatabase,
     OutputManagerSqliteDatabase,
     ContactsServiceSqliteDatabase,
 >;
+
+pub type WalletThreadSafe = Arc<RwLock<WalletSqlite>>;
```

Comment:
I don't think we need this. The WalletSqlite struct is a handle that is meant to be cloneable and its interfaces are async and threadsafe. I believe we can just copy the wallet handle directly for each component of this app (grpc server gets a clone of the wallet object and the other component, TUI, script etc, gets its own copy).

Taking this out means we remove the chance for weird deadlocks and also don't need the extra boiler plate of asking for a lock before interacting with the wallet.

Reply:
As above, I'll try this out

Reply:
Discussed offline, this is going to require a bit more of a refactoring due to the shutdown signals from various clients